File: frontend/src/main/java/ru/mirea/cnoss/service/node/NodeService.java
package ru.mirea.cnoss.service.node;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestHeader;
import ru.mirea.cnoss.service.node.dto.DetailedNodeResponse;


@FeignClient(url = "http://localhost:8085/", value = "nodes")
public interface NodeService {

    @GetMapping("/network-operational-support-system-api/node/{id}")
    DetailedNodeResponse getNodeById(@PathVariable("id") String id, @RequestHeader("Authorization") String authorization);
}

File: frontend/src/main/frontend/services/NodeService.ts
import { NodeEndpoint } from 'Frontend/generated/endpoints';
import DetailedNodeResponse from 'Frontend/generated/ru/mirea/cnoss/service/node/dto/DetailedNodeResponse'

export async function getNodeById(id: string): Promise<DetailedNodeResponse> {
  return await NodeEndpoint.getNode(id);
}